include/boost/json/impl/storage_ptr.ipp

```diff
@@ -15,6 +15,16 @@
 BOOST_JSON_NS_BEGIN
 
 // KRYSTIAN NOTE: we need P1077
+#ifdef __clang__
```

Comment:
P1077 won't help us until what, C++23? Or C++26? Or never? And even then, we still have to support older versions of C++. Therefore, P1077 will never be of any help.